src/lib/storage/chunk.hpp

```diff
@@ -25,9 +25,15 @@ class Chunk : private Noncopyable {
   // Creates an empty chunk.
   Chunk() = default;
 
+  // Creates a chunk with the segments vector initialized to the specified capacity
+  explicit Chunk(ColumnID n_columns);
+
   // Adds a segment to the "right" of the chunk.
   void add_segment(const std::shared_ptr<AbstractSegment> segment);
 
+  // Exchanges a segment at the specified index
+  void add_segment_at(const std::shared_ptr<AbstractSegment> segment, const ColumnID position);
```

Comment:
Method name can be confusing as it sounds as if you are inserting a segment (instead of swapping)